lib/omniauth/strategies/openid_connect/backchannel_logout.rb

```diff
@@ -0,0 +1,70 @@
+module OmniAuth
+  module Strategies
+    class OpenIDConnect
+      module BackchannelLogout
+        def self.prepended(base)
+          base.class_exec do
+            # Callback for the backchannel logout
+            option :backchannel_logout_callback, nil
+          end
+        end
+
+        def other_phase
+          if on_backchannel_flow?
+            return handle_backchannel_flow
+          end
+
+          super
+        end
+
+        def on_backchannel_flow?
+          return false unless options.backchannel_logout_callback
+
+          backchannel_logout_path_pattern.match?(current_path)
+        end
+
+        def handle_backchannel_flow
+          discover!
+          perform_backchannel_logout!(params['logout_token'])
+          backchannel_response
+        rescue StandardError => e
+          Rack::Response.new(
+            [e.message],
+            400,
+            {}
```

Comment:
We ought to use the no-cache header here too!

Reply:
Thanks, done!